rig-qdrant/examples/qdrant_vector_search.rs

```diff
@@ -0,0 +1,80 @@
+// To run this example:
+// cargo run --release --example qdrant_vector_search
+
+use std::env;
+
+use qdrant_client::{
+    qdrant::{
+        CreateCollectionBuilder, Distance, PointStruct, QueryPointsBuilder, UpsertPointsBuilder,
+        VectorParamsBuilder,
+    },
+    Payload, Qdrant,
+};
+use rig::{
+    embeddings::EmbeddingsBuilder,
+    providers::openai::{Client, TEXT_EMBEDDING_ADA_002},
+    vector_store::VectorStoreIndex,
+};
+use rig_qdrant::QdrantVectorStore;
+use serde_json::json;
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    const COLLECTION_NAME: &str = "rig-collection";
+
+    let client = Qdrant::from_url("http://localhost:6334").build()?;
+
+    // Create a collection with 1536 dimensions if it doesn't exist
+    if !client.collection_exists(COLLECTION_NAME).await? {
+        client
+            .create_collection(
+                CreateCollectionBuilder::new(COLLECTION_NAME)
+                    .vectors_config(VectorParamsBuilder::new(1536, Distance::Cosine)),
+            )
+            .await?;
+    }
+
+    // Initialize OpenAI client.
+    // Get your API key from https://platform.openai.com/api-keys
+    let openai_api_key = env::var("OPENAI_API_KEY").expect("OPENAI_API_KEY not set");
+    let openai_client = Client::new(&openai_api_key);
+
+    let model = openai_client.embedding_model(TEXT_EMBEDDING_ADA_002);
+
+    let documents = EmbeddingsBuilder::new(model.clone())
+        .simple_document("0981d983-a5f8-49eb-89ea-f7d3b2196d2e", "Definition of a *flurbo*: A flurbo is a green alien that lives on cold planets")
```

Comment:
It is not clear to me why the user needs to generate a UUID to use when adding documents here. Are document ids enforced as UUIDs by Qdrant? it's a bit clumbersome in the codebase, I think it could be abstracted away in some form.

Reply:
Otherwise, is something like the [uuid](https://crates.io/crates/uuid) crate recommended for this? Perhaps generating it off of a normal id.

Reply:
Yes.
Qdrant only allows unsigned ints and UUIDs as point IDs.
https://qdrant.tech/documentation/concepts/points/#point-ids

Reply:
To keep the example clear, I've opted to use literal IDs here.